src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -693,10 +693,11 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
                     cmd, vnet_name, location, tags, vnet_address_prefix, subnet, subnet_address_prefix))
 
         if nsg_type == 'new':
-            nsg_rule_type = 'rdp' if os_type.lower() == 'windows' else 'ssh'
+            if nsg_rule is None:
+                nsg_rule = 'RDP' if os_type.lower() == 'windows' else 'SSH'
```

Comment:
why not keep original lower case (rdp, ssh) ?

Reply:
Don't care about old code. Abandon it.
https://en.wikipedia.org/wiki/Secure_Shell
https://docs.microsoft.com/en-us/windows/win32/termserv/remote-desktop-protocol